src/main/java/com/aws/greengrass/config/ConfigurationReader.java

```diff
@@ -166,7 +219,30 @@ public static boolean validateTlog(Path tlogPath) {
      */
     public static Configuration createFromTLog(Context context, Path p) throws IOException {
         Configuration c = new Configuration(context);
-        ConfigurationReader.mergeTLogInto(c, p);
+        createFromTLog(context, p, ConfigurationMode.WITH_VALUES);
         return c;
     }
+
+    /**
+     * Create a Configuration based on a transaction log's path.
+     *
+     * @param context           root context for the configuration
+     * @param p                 path to the transaction log
+     * @param configurationMode Configuration mode
+     * @return Configuration from the transaction log
+     * @throws IOException if reading the transaction log fails
+     */
+    public static Configuration createFromTLog(Context context, Path p, ConfigurationMode configurationMode)
+            throws IOException {
+        Configuration c = new Configuration(context);
+        mergeTLogInto(c, p, configurationMode);
+        return c;
+    }
+
+    enum ConfigurationMode {
```

Comment:
this is needed to call a public API, so it should be public.

Reply:
I made the API private instead because for now I don't see the need to make it public